pkg/combustion/templates/11-time-setup.sh.tpl

```diff
@@ -16,3 +16,31 @@ echo "pool {{ . }} iburst" >> /etc/chrony.d/eib-sources.conf
 {{ range .Servers -}}
 echo "server {{ . }} iburst" >> /etc/chrony.d/eib-sources.conf
 {{ end -}}
+
+{{ if .ForceWait -}}
+cat <<EOF >/etc/systemd/system/firstboot-timesync.service
+[Unit]
+Description=Attempt NTP timesync to occur before starting Kubernetes services
+Requires=chronyd.service
+Wants=network-online.target
+After=network-online.target
+After=chrony-wait.service
+Before=rke2-server.service
+Before=rke2-agent.service
+Before=k3s.service
+
+[Service]
+User=root
+Type=oneshot
+ExecStart=/usr/bin/echo "[INFO] Either reached 180s timeout or was successful in timesync before starting system services."
+RemainAfterExit=true
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+systemctl enable chrony-wait
+systemctl enable firstboot-timesync.service
+
+echo "[WARN]: Waiting up to 180s to synchronise system clock with available NTP sources."
```

Comment:
This will log the message as part of the Combustion unit / phase. Isn't it better to include it as a `ExecStartPre` of the firstboot-timsync unit instead?

Reply:
@atanasdinov the reason why we have this here is because the boot will pause and ideally we need something on the console to explain what's happening. Because the `firstboot-timesync` unit depends on `chrony-wait` I'm not sure that `ExecStartPre` will run until chrony has synchronised. I can check.

Reply:
So, you can't use `ExecStartPre` with a `oneshot` service. Unfortunately I think this is as good as we're going to get.